test/kad-dht.js

```diff
@@ -1,112 +1,692 @@
+/* eslint max-nested-callbacks: ["error", 6] */
 /* eslint-env mocha */
 'use strict'
 
 const chai = require('chai')
 const dirtyChai = require('dirty-chai')
+const chaiBytes = require('chai-bytes')
 const expect = chai.expect
 chai.use(dirtyChai)
+chai.use(chaiBytes)
 
-const series = require('async/series')
 const crypto = require('crypto')
+const each = require('async/each')
 const parallel = require('async/parallel')
-const waterfall = require('async/waterfall')
-const bl = require('bl')
+const times = require('async/times')
+
+const waitFor = require('./utils/wait-for-fn')
 
 const DaemonFactory = require('ipfsd-ctl')
+const IPFS = require('ipfs')
 const goDf = DaemonFactory.create()
-const jsDf = DaemonFactory.create({ type: 'js' })
-
-describe.skip('kad-dht', () => {
-  describe('a JS node in the land of Go', () => {
-    let jsD
-    let goD1
-    let goD2
-    let goD3
-
-    before((done) => {
-      parallel([
-        (cb) => goDf.spawn({ initOptions: { bits: 1024 } }, cb),
-        (cb) => goDf.spawn({ initOptions: { bits: 1024 } }, cb),
-        (cb) => goDf.spawn({ initOptions: { bits: 1024 } }, cb),
-        (cb) => jsDf.spawn({ initOptions: { bits: 512 } }, cb)
-      ], (err, nodes) => {
+const jsDf = DaemonFactory.create({ type: 'proc', exec: IPFS })
+
+const getConfig = (bootstrap) => ({
+  Bootstrap: bootstrap,
+  Discovery: {
+    MDNS: {
+      Enabled: false
+    },
+    webRTCStar: {
+      Enabled: false
+    }
+  }
+})
+
+const spawnGoDaemon = (bootstrap = [], callback) => {
+  goDf.spawn({
+    initOptions: { bits: 1024 },
+    config: getConfig(bootstrap)
+  }, callback)
+}
+
+const spawnJsDaemon = (bootstrap = [], callback) => {
+  jsDf.spawn({
+    initOptions: { bits: 512 },
+    config: getConfig(bootstrap),
+    libp2p: {
+      config: {
+        dht: {
+          enabled: true
+        }
+      }
+    }
+  }, callback)
+}
+
+const getBootstrapAddr = (node, callback) => {
+  node.api.id((err, res) => {
+    expect(err).to.not.exist()
+    expect(res.id).to.exist()
+
+    callback(null, res.addresses[0])
+  })
+}
```

Comment:
This function is a tad misleading. All it does is extract the first listing address of a node. Perhaps rename to getNodeAddr?

Reply:
Yeah, agree! I will change that!